zio-cli/shared/src/main/scala/zio/cli/PrimType.scala

```diff
@@ -36,20 +37,93 @@ object PrimType {
   }
 
   final case class Path(pathType: PathType, exists: Boolean) extends PrimType[JPath]
-  case object Text                                           extends PrimType[String]
-  case object Decimal                                        extends PrimType[BigDecimal]
-  case object Integer                                        extends PrimType[BigInt]
-  case object Instant                                        extends PrimType[JInstant]
-  case object LocalDate                                      extends PrimType[JLocalDate]
-  case object LocalDateTime                                  extends PrimType[JLocalDateTime]
-  case object LocalTime                                      extends PrimType[JLocalTime]
-  case object MonthDay                                       extends PrimType[JMonthDay]
-  case object OffsetDateTime                                 extends PrimType[JOffsetDateTime]
-  case object OffsetTime                                     extends PrimType[JOffsetTime]
-  case object Period                                         extends PrimType[JPeriod]
-  case object Year                                           extends PrimType[JYear]
-  case object YearMonth                                      extends PrimType[JYearMonth]
-  case object ZonedDateTime                                  extends PrimType[JZonedDateTime]
-  case object ZoneId                                         extends PrimType[JZoneId]
-  case object ZoneOffset                                     extends PrimType[JZoneOffset]
+  {
+    import PathType._
+    def validate(value: String): IO[String,JPath] = {
+      val path = JPaths.get(value)
+      val (matchesType, existsCheck) = pathType match {
+        case Anything => (true, JFiles.exists(path))
+        case File => (JFiles.isDirectory(path), JFiles.exists(path))
+        case Directory => (JFiles.isDirectory(path), JFiles.exists(path))
+      }
+      if (matchesType && exists == existsCheck) IO.succeed(path) 
+      else IO.fail(s"Specified path '$value' does not match requirements.")
+    }
+  }
+  
+  case object Text extends PrimType[String] {
+    def validate(value: String): IO[String, String] = attempt(value, _ => value, "text")
+  }
+  
+  case object Decimal extends PrimType[BigDecimal] {
+    def validate(value: String): IO[String, BigDecimal] = attempt(value, BigDecimal(_), "decimal")
+  }
+  
+  case object Integer extends PrimType[BigInt] {
+    def validate(value: String): IO[String, BigInt] = attempt(value, BigInt(_), "integer")
+  }
+
+  case object Boolean extends PrimType[Boolean] {
```

Comment:
How do we want to handle Boolean?

Reply:
Maybe also support on / off, yes / no, 1 / 0???

Reply:
I covered those extra cases.